core/src/main/kotlin/com/malinskiy/marathon/execution/progress/tracker/PoolProgressTracker.kt

```diff
@@ -42,33 +42,36 @@ class PoolProgressTracker {
         }
     }
 
-    private fun updateStatus(test: Test, newStatus: ProgressEvent) {
-        tests[test]?.transition(newStatus)
-    }
+    private fun updateStatus(test: Test, newStatus: ProgressEvent) = tests[test]?.transition(newStatus)
 
     private val totalTests = AtomicInteger(0)
     private val completed = AtomicInteger(0)
     private val failed = AtomicInteger(0)
 
-    fun testStarted(test: Test, device: DeviceInfo) {
+    fun testStarted(test: Test) {
         tests.computeIfAbsent(test) { _ -> createState() }
     }
 
-    fun testFailed(test: Test, device: DeviceInfo) {
+    fun testFailed(test: Test) {
+        if(tests[test]?.state == ProgressTestState.Passed) {
+            //Return early because the test already passed
+            return
+        }
+
         updateStatus(test, ProgressEvent.Failed)
         failed.updateAndGet {
             it + 1
         }
     }
 
-    fun testPassed(test: Test, device: DeviceInfo) {
+    fun testPassed(test: Test) {
         completed.updateAndGet {
             it + 1
```

Comment:
You might want to add an early exit here as well to account for multiple successful executions

Reply:
This will not work as expected since multiple success executions of tests are actually not affecting our percentage